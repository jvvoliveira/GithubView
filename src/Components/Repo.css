.repo{
    display: block;
    margin: 30px;
    padding: 10px;
    border: solid;
}