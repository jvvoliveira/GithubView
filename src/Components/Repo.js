import React from "react";
import styles from "./Repo.module.scss";

const Repo = props => {
  return (
    <a className={styles.repositorio} href={props.link} target="_blank">
      <h3 data-testid='name'>{props.name}</h3>
      <div className={styles.text}>
        <p data-testid='description'>{props.description}</p>
      </div>
      <p>{props.language}</p>
    </a>
  );
};

export default Repo;
