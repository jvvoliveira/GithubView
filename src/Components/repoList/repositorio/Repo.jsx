import React from "react";
import styles from "./Repo.module.scss";

const Repo = props => {
  const { name, language, description, link, index } = props;
  return (
    <a
      data-testid={`repo-${index}`}
      className={styles.repositorio}
      href={link}
      target="_blank"
      key={index}
    >
      <h3 data-testid="name">{name}</h3>
      <div className={styles.text}>
        <p data-testid="description">{description}</p>
      </div>
      <p>{language}</p>
    </a>
  );
};

export default Repo;
