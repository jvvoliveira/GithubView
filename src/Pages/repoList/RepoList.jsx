import React, { useContext } from "react";
import Repositorio from "./repositorio/Repositorio";
import { loadingContext } from "../../App";
import styles from "./RepoList.module.scss";
import { getReposByUser } from "../../services";
import EmptyContent from "../../Components/emptyContent/EmptyContent";

const moreItens = async (usuario, page, setRepos, repos, setPage) => {
  try {
    const responseRepos = await getReposByUser(usuario.data.login, page);
    setRepos([...repos.data, ...responseRepos.data], "OK");
    setPage(page + 1);
  } catch (error) {
    console.log(error);
  }
};

const ViewMore = (props) => {
  const { page, pages, usuario, setRepos, repos, setPage } = props;
  const _moreItens = () => {
    moreItens(usuario, page, setRepos, repos, setPage);
  };

  if (page > pages)
    return (
      <div data-testid="allRepos" className={styles.viewMore}>
        <h3>Esses são todos os repositórios públicos</h3>
      </div>
    );

  return (
    <div
      data-testid="viewMore"
      className={styles.viewMore}
      onClick={_moreItens}
    >
      <h3>Ver mais</h3>
    </div>
  );
};

const RepoList = () => {
  const { repos, page, usuario, setRepos, setPage, repos_max } = useContext(
    loadingContext
  );
  let pages = repos_max / 8;
  if (repos_max % 8 > 0) {
    pages++;
  } //8 repositórios por página

  const list = repos.data.map((repo, index) => {
    return (
      <Repositorio
        name={repo.name}
        link={repo.html_url}
        language={repo.language}
        description={repo.description}
        index={index}
        key={index}
      />
    );
  });

  if (repos.status !== "OK")
    return <EmptyContent repos dataTestId="semRepositorio" message="Repositórios não encontrados" />

  if (repos.data.length === 0)
    return <EmptyContent repos dataTestId="repositoriosVazio" message="Usuário sem repositórios públicos" />

  return (
    <div className={styles.lista_infinita}>
      {list}
      <ViewMore
        page={page} pages={pages} usuario={usuario}
        setRepos={setRepos} repos={repos} setPage={setPage}
      />
    </div>
  );
};

export default RepoList;
