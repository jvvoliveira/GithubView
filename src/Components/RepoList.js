import React, { useContext } from 'react';
import Repo from './Repo';
import { loadingContext } from '../App';

export default () => {

    const { data, status } = useContext(loadingContext).repos;


    if (status == 'OK') {
        if (data.length !== 0) {
            return (
                data.map(repo => {
                    return (
                        <Repo name={repo.name} link={repo.html_url} language={repo.language}
                            description={repo.description}></Repo>
                    );
                })
            );
        } else {
            return (
                <h2>Usuário sem repositórios públicos</h2>
            );
        }
    }
    else{
        return(
            <h2>Repositórios não encontrados</h2>
        )
    }
}
