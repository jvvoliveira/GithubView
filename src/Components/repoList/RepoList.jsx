import React, { useContext } from "react";
import Repo from "./repositorio/Repo";
import { loadingContext } from "../../App";
import styles from "./RepoList.module.scss";

export const moreItens = (usuario, page, setRepos, repos, setPage) => {
  fetch(
    `https://api.github.com/users/${usuario.data.login}/repos?per_page=8&page=${page}`
  )
    .then(response => response.json())
    .then(value => {
      if (!value.message) {
        setRepos([...repos.data, ...value], "OK");
      } else {
        setRepos([], value.message);
      }
    });
  setPage(page + 1);
};

export const viewMore = (page, pages, usuario, setRepos, repos, setPage) => {
  const _moreItens = () => {
    moreItens(usuario, page, setRepos, repos, setPage);
  };
  if (page < pages) {
    return (
      <div className={styles.viewMore} onClick={_moreItens}>
        <h3>Ver mais</h3>
      </div>
    );
  } else {
    return (
      <div className={styles.viewMore}>
        <h3>Esses são todos os repositórios públicos</h3>
      </div>
    );
  }
};

const RepoList = () => {
  const { repos, page, usuario, setRepos, setPage, repos_max } = useContext(
    loadingContext
  );
  console.log(repos_max);
  let pages = repos_max / 8;
  if (repos_max % 8 > 0) {
    pages++;
  } //8 repositórios por página
  const list = repos.data.map(repo => {
    return (
      <Repo
        name={repo.name}
        link={repo.html_url}
        language={repo.language}
        description={repo.description}
        key={repo.html_url}
      />
    );
  });

  if (repos.status === "OK") {
    if (repos.data.length !== 0) {
      return (
        <div className={styles.lista_infinita}>
          {list}
          {viewMore(page, pages, usuario, setRepos, repos, setPage)}
        </div>
      );
    } else {
      return <h2>Usuário sem repositórios públicos</h2>;
    }
  } else {
    return <h2>Repositórios não encontrados</h2>;
  }
};

export default RepoList;
