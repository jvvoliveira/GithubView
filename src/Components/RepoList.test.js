import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import RepoList from './RepoList';
import Repo from './Repo';

describe("test initial Repolist", () => {
    it("should renders without crashing", () => {
        const repos = [
            {
                name: 'name',
                language: 'language',
                description: 'description',
                html_url: 'link'
            }
        ]
        const div = document.createElement('div');
        ReactDOM.render(<RepoList repos={repos} />, div);
        ReactDOM.unmountComponentAtNode(div);
    });
    it("snapshot", () => {
        const repos = [
            {
                name: 'name',
                language: 'language',
                description: 'description',
                html_url: 'link'
            },
            {
                name: 'name2',
                language: 'language2',
                description: 'description2',
                html_url: 'link2'
            },
            {
                name: 'name3',
                language: 'language3',
                description: 'description3',
                html_url: 'link3'
            }
        ];
        const tree = renderer.create(<RepoList repos={repos}></RepoList>).toJSON();
        expect(tree).toMatchSnapshot();
    });
})