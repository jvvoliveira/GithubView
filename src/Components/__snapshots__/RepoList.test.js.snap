// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test initial Repolist snapshot 1`] = `
Array [
  <div
    className="repo"
  >
    <h3>
      name
    </h3>
    <p>
      description
    </p>
    <a
      href="link"
      target="_blank"
    >
      repositório
    </a>
    <p>
      Linguagem: 
      language
    </p>
    <br />
  </div>,
  <div
    className="repo"
  >
    <h3>
      name2
    </h3>
    <p>
      description2
    </p>
    <a
      href="link2"
      target="_blank"
    >
      repositório
    </a>
    <p>
      Linguagem: 
      language2
    </p>
    <br />
  </div>,
  <div
    className="repo"
  >
    <h3>
      name3
    </h3>
    <p>
      description3
    </p>
    <a
      href="link3"
      target="_blank"
    >
      repositório
    </a>
    <p>
      Linguagem: 
      language3
    </p>
    <br />
  </div>,
]
`;
